src/classes/CW_BriefApprovalProcess.cls

```diff
@@ -0,0 +1,167 @@
+	 				errorMessage = ConstantUtil.SUBMIT_FOR_APPROVAL_BRIEF_NO_ROADMAP;
+			 		submitBriefForApproval = false;
+	 			}
+ 			}
+ 			// campaign IS urgent
+ 			else 
+ 				submitBriefForApproval = true;
+ 		}
+ 		else 
+ 		{
+ 			//NOT all the required fields are populated Brief CANNOT be submitted
+ 			submitBriefForApproval = false;
+ 		}
+ 		
+		if (getapproverIds().isEmpty())
+		{
+			submitBriefForApproval = false;
+			errorMessage = ConstantUtil.SUBMIT_FOR_APPROVAL_NO_BRIEF_APPROVER;
+		} 
+ 		
+ 		return submitBriefForApproval;
+ 	}
+ 	
+ 	//get all the field from fieldset
+	public Set<String> getBriefFieldset() {
+		Set<String> fnames = new set<string>();         
+    	
+    	for(Schema.FieldSetMember f : Schema.SObjectType.CW_Campaign__c.FieldSets.Marcom_Brief_Fields.getFields()) {
+           fnames.add(f.getFieldPath());
+    	}
+       return fnames; 
+    }
+    
+    // query all the campaigns related to the roadmap
+    public List<CW_Campaign__c> getBriefRequiredFields(Id campId)
+    {
+    	briefFieldSet = getBriefFieldset();
+		
+		String queryString = 'SELECT ';		
+		
+		for (String curr: briefFieldset){
+			queryString += curr + ', ';
+			
+		}
+		queryString = queryString.removeEnd(', ');
+        queryString += ' FROM CW_Campaign__c WHERE Id =\'' + campId + '\'';
+        return Database.query(queryString);
+    }
+ 	
+ 	//if submit brief criteria is met submit for approval
+	public Boolean submitBriefForApproval(CW_Campaign__c camp) 
+    {
+    	Boolean isSubmitted = false;
+
+    	if (checkBriefForApproval(camp))
+    	{
+    		camp.Brief_Approval__c = true;
+    		upsert camp;
+	    	Approval.ProcessSubmitRequest req = new Approval.ProcessSubmitRequest();
+	 		req.setObjectId(camp.Id);
+		 	req.setNextApproverIds(getapproverIds());
+		 	Approval.ProcessResult result = Approval.process(req);
+				
+		 		if(result.isSuccess()) 
+		 		{ 
+		 			//CW_LockFieldsApprovalSubmission lockBriefFileds = new CW_LockFieldsApprovalSubmission();
+		 			//lockBriefFileds.lockFields(ConstantUtil.MARCOM_BRIEF_FIELDSET_NAME,ConstantUtil.MARCOM_BRIEF_LOCKED_FIELDSET_NAME);
+		 			ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.CONFIRM, ConstantUtil.SUBMIT_FOR_APPROVAL_SUCCESFULLY)); 
+		 			isSubmitted = true;
+		 			
+		 		} 
+		 		else { 
+		 			ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, ConstantUtil.SUBMIT_FOR_APPROVAL_FAILED)); 
+		 		} 
+    	}
+    	//notify user why Brief CANNOT be submitted
+ 		else if (errorMessage!= null)
+ 			ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error,errorMessage));
+    	
+    	return isSubmitted;
+    }
+    
+    // get the approverId from custom settings
+ 	public static List<Id> getApproverIds()
+ 	{
+ 		List<ApproverListSetting__c> briefApprover = new List<ApproverListSetting__c>();
+ 		
+ 		List<Id> briefApproverIds = new List<Id>();
+		
+		//get approver for Roadmap
+ 		briefApprover = [SELECT Role__c, Username__c, Type__c 
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.